### ISSUE TITLE
Service cannot registered, openAPI query result not consistent

### ISSUE THREAD

**Issue by evancyz** [alibaba/nacos#1712]
hello ,
I use nacos-sync to sync zkRegister to nacoRsegister

when i sync one service , in the nacos op, it looks like right
![image](https://user-images.githubusercontent.com/22286097/63212785-fe963980-c13b-11e9-82cb-5188d3d8ba9c.png)

but, when i use consumer, it can not find provider

i use openAPI try to solve question：
![image](https://user-images.githubusercontent.com/22286097/63212827-83815300-c13c-11e9-8f36-0809f54a34c3.png)
when i use this command, it looks like ok

bug when i use this command
![image](https://user-images.githubusercontent.com/22286097/63212877-0f937a80-c13d-11e9-863e-3fb9f549a0b4.png)
hosts is empty。
Was the registration successful？

**Comment by evancyz** [comment]
when my consumer start,  in nacos backgroup it likes that 。
then the dubbo error， can not find provider

![image](https://user-images.githubusercontent.com/22286097/63212953-cb54aa00-c13d-11e9-962e-9114362a034b.png)

**Comment by evancyz** [comment]
I found the question on my IDE console:
request for registration like  that : 
http://xxxxx:8848/nacos/v1/ns/instance/list?healthyOnly=false&namespaceId=public&clientIP=192.168.1.8&serviceName=DEFAULT_GROUP@@providers:com.xxx.xxx.api.service.xxxService:1.0.0:&udpPort=61054&encoding=UTF-8

 com.xxx.xxx.api.service.xxxService:1.0.0: 

in the last, it has the extra  ":" 
(:

**Comment by evancyz** [comment]
I found the question is on：

org.apache.dubbo.registry.nacos.getServiceName  
use the util : org.apache.dubbo.common.getColonSeparatedKey to generate consumer 

if the group name is blank 
in the end, it will append the " : "

(:
this is the bug ? i dont have the groupName in the old project , it not process well i guess 
![image](https://user-images.githubusercontent.com/22286097/63213781-fdb4d600-c142-11e9-95a5-0e3c6e826584.png)

**Comment by evancyz** [comment]
my dubbo provider version ： 2.6.5
in old dubbo version，the group or version can be null
in the  new version, the rule is changed? 

i look the test,it seems like group extra ":" is right 
`
    @Test
    public void testGetColonSeparatedKey() {
        URL url1 = URL.valueOf("10.20.130.230:20880/context/path?interface=org.apache.dubbo.test.interfaceName&group=group&version=1.0.0");
        Assertions.assertEquals("org.apache.dubbo.test.interfaceName:1.0.0:group", url1.getColonSeparatedKey());

        URL url2 = URL.valueOf("10.20.130.230:20880/context/path?interface=org.apache.dubbo.test.interfaceName&version=1.0.0");
        Assertions.assertEquals("org.apache.dubbo.test.interfaceName:1.0.0:", url2.getColonSeparatedKey());

        URL url3 = URL.valueOf("10.20.130.230:20880/context/path?interface=org.apache.dubbo.test.interfaceName&group=group");
        Assertions.assertEquals("org.apache.dubbo.test.interfaceName::group", url3.getColonSeparatedKey());

        URL url4 = URL.valueOf("10.20.130.230:20880/context/path?interface=org.apache.dubbo.test.interfaceName");
        Assertions.assertEquals("org.apache.dubbo.test.interfaceName::", url4.getColonSeparatedKey());
    }
`

**Comment by evancyz** [comment]
i search the answer in `nacos-sync` project

nacos-sync use nacos `NamingService` to register service

in the ZookeeperSyncToNacosServiceImpl#getServiceNameFromCache，the method is generate the service name

the code display：
```
    /**
     * cteate Dubbo service name
     *
     * @param taskId task id
     * @param queryParam dubbo metadata
     */
    protected String getServiceNameFromCache(String taskId, Map<String, String> queryParam) {
        return nacosServiceNameMap
                .computeIfAbsent(taskId, (key) -> Joiner.on(":").skipNulls().join(CATALOG_KEY,
                        queryParam.get(INTERFACE_KEY), queryParam.get(VERSION_KEY),
                        queryParam.get(GROUP_KEY)));
    }

```
in the nacos-sync entity `task`，the dubbo group is ""，not the null
but，above the method，the field  `Map<String, String> queryParam` is the zookeeper entity，so the GROUP_KEY is null。

this is the answer （：

i will give the `nacos-sync` issue （：

but i think the “:” in the end  is not pretty

Thank you！

**Comment by chuntaojun** [comment]
We will go to repair the nacos - sync components problem

**Comment by paderlol** [comment]
``` java
// 这是 2.7.5
String parameterValue = this.getParameter(parameterName);
        if (!StringUtils.isBlank(parameterValue)) {
            if (!first) {
                target.append(":");
            }
            target.append(parameterValue);
        } else {
            target.append(":");
        }
```
``` java
// 2.6.x

    private String getServiceName(URL url, String category) {
        StringBuilder serviceNameBuilder = new StringBuilder(category);
        appendIfPresent(serviceNameBuilder, url, Constants.INTERFACE_KEY);
        appendIfPresent(serviceNameBuilder, url, Constants.VERSION_KEY);
        appendIfPresent(serviceNameBuilder, url, Constants.GROUP_KEY);
        return serviceNameBuilder.toString();
    }

    private void appendIfPresent(StringBuilder target, URL url, String parameterName) {
        String parameterValue = url.getParameter(parameterName);
        if (!StringUtils.isBlank(parameterValue)) {
            target.append(SERVICE_NAME_SEPARATOR).append(parameterValue);
        }
    }

```
This is a Dubbo version issue, Without about a problem with Nacos Sync. I consider if there is a way to solve this problem.

**Comment by paderlol** [comment]
This issue has been resolved, you can use version 0.3.5 of Nacos Sync.
[PR](https://github.com/nacos-group/nacos-sync/pull/132)
